services/api/src/api/routes/rows.py

```diff
@@ -67,19 +67,27 @@ class ParquetDataProcessingError(Exception):
 
 # TODO: how to invalidate the cache when the parquet branch is created or deleted?
 @lru_cache(maxsize=128)
```

Comment:
I didn't change the use of `lru_cache` here (in rows.py), but I noticed it is not used in first_rows_from_parquet.py.

Is there any reason for this? Also note that now the returned fs will always be the same as it no longer depends on the dataset name...

Reply:
you're right, we could use it too in the worker. Possibly it was more crucial here: the worker works asynchronously, while the code here is called synchronously when doing a request on /rows.